src/test/java/uk/nhs/ctp/service/ValidationServiceTest.java

```diff
@@ -51,4 +58,164 @@ public void zip_creation() throws IOException {
 //    }
 //    System.out.println("Output written to " + zipFile);
   }
+
+  @Test
+  public void zipAudits_withMoreThan1ServiceSearchAudit_shouldFail() throws IOException {
+    var audits = List.of(AuditSession.builder().build(), AuditSession.builder().build());
+
+    expectedException.expect(UnsupportedOperationException.class);
+    validationService.zipAudits(audits, OperationType.SERVICE_SEARCH);
+  }
+
+  @Test
+  public void zipAudits_withMoreThan1ServiceSearchAuditEntry_shouldFail() throws IOException {
+    var audits = List.of(AuditSession.builder()
+        .entry(AuditEntry.builder().build())
+        .entry(AuditEntry.builder().build())
+        .build());
+
+    expectedException.expect(UnsupportedOperationException.class);
+    validationService.zipAudits(audits, OperationType.SERVICE_SEARCH);
+  }
+
+  @Test
+  public void zipAudits_withEncountersLackingCaseId_shouldFail() throws IOException {
+    var audits = List.of(
+        AuditSession.builder().build(),
+        AuditSession.builder().additionalProperty("caseId", "validCaseId").build());
+
+    expectedException.expect(UnsupportedOperationException.class);
+    validationService.zipAudits(audits, OperationType.ENCOUNTER);
+  }
+
+  @Test
+  public void zipAudits_withEncounterLackingCaseId_shouldFail() throws IOException {
+    var audits = List.of(AuditSession.builder().build());
+
+    expectedException.expect(UnsupportedOperationException.class);
+    validationService.zipAudits(audits, OperationType.ENCOUNTER);
+  }
+
+  @Test
+  public void zipAudits_shouldOnlyZipGetEntries() throws IOException {
```

Comment:
As above, we also want POSTs, and potentially the outer audit in some cases, not just the entries

Reply:
Agreed on the outer audits, will do.
Regarding `POST`s, since we want to capture both response and request bodies I propose a change in path formatting for all requests:
`(service_search|encounter{encounterId})/full/url/path.(request|response).(json|xml)`
With this format I'm assuming GET requests will always end in `response.(json|xml)`, while POST requests will always have both a `request.(json|xml)` and a `response.(json|xml)`, even when the body is empty, preserving current behaviour?

Reply:
👍  Yep, agreed, with one addition - as POSTs are not idempotent, we probably also want to add a sequence ID/counter to each post request to avoid name collisions. Not sure what happens at the moment if the same GET request is made twice - presumably the .zip file contains multiple entries for the same path? Is that valid?

Reply:
Currently the zip will only contain the first response to a GET request to a requestUrl within a particular encounterId (using a `HashSet` on the zip's inner filenames).

Reply:
Should we extend the format for all requests to the following?
`(service_search|encounter{encounterId})/full/url/path.`**`{sequenceCounter}`**`.(request|response).(json|xml)`

Reply:
That makes sense I think, yes.